cpp/src/arrow/compute/kernels/vector_sort_test.cc

```diff
@@ -675,6 +687,15 @@ TYPED_TEST(TestArraySortIndicesForDecimal, DecimalSortTestTypes) {
                           "[1, 5, 3, 0, 2, 4]");
 }
 
+TEST(TestArraySortIndices, Null) {
+  for (const auto null_placement : AllNullPlacements()) {
+    for (const auto order : AllOrders()) {
+      AssertSortIndices(null(), "[null, null, null, null]", order, null_placement,
+                        "[0, 1, 2, 3]");
+    }
+  }
+}
+
```

Comment:
Can you also add tests for chunked array, record batch and table sorting?

Reply:
Ah, good point. I've added the necessary support for those cases.